src/Parse/Markdown.php

```diff
@@ -90,16 +105,19 @@ public function parseLine($text)
     /**
      * Internal method for parsing
      */
-    protected function parseInternal($text, $method = 'text')
+    protected function parseInternal($text, $method = 'text', Parsedown $parser = null)
     {
+        if (is_null($parser)) {
+            $parser = $this->getParser();
+        }
         $data = new MarkdownData($text);
 
-        $this->fireEvent('beforeParse', $data, false);
-        Event::fire('markdown.beforeParse', $data, false);
+        $this->fireEvent('beforeParse', [$data], false);
+        Event::fire('markdown.beforeParse', [$data], false);
```

Comment:
Does this cause any issues having it changed to an array instead of just the variable?

Reply:
It is listed as one of the breaking changes. Since event payloads are processed into an array, this was done mainly to appease PHPStan, but it's probably something we should stick to with events anyway.